src/components/displaySvg/finalSvgDisplay.js

```diff
@@ -0,0 +1,157 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import DisplaySvg from "./displaySvg";
+import finalSvgDisplay from "./finalSvgDisplay.scss";
+import { connect } from "react-redux";
+import base64 from "base-64";
+import { SVG_TAG_NAMES } from "../../constants/constants";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+import { wrapElements, updateSvgElements } from "../functions";
+
+class FinalSvgDisplay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      wrappedPathsElement: [],
+      pathArrayState: null,
+      storePathsIds: [],
+      clickedSvgState: []
+    };
+  }
+
+  downloadNewSvgFiles = () => {
```

Comment:
please clarify what this function does please

Reply:
This downloads new svg , the modified ones by clicking

Reply:
this function downloads newly created svg by clicking  any svg into a folder so I do not get what other name would you want me to use.

Reply:
Is this to download optimized svgs or the svgs with the className definitions?

Reply:
Optimised ones and edited ones if exist.

Reply:
so it would downloadOptimizedSvgs ?

Reply:
yes

Reply:
so you still cant think of a better name for the function that will downloadOptimizedSvgs than "downloadNewSvgFiles"?

Reply:
Thats your preference of naming, because NewSvg are Optimised svg.. these naming are just the samething. I will change to your preference then.

Reply:
So, they're not the same thing. New is severely undescriptive. Optimized tells us exactly what changed about the svg. For instance, if you had 3 functions that returned new svgs, one that adds the defs tag with the styling for the svg, another that optimizes it and another with the colours of the paths changed, how would you know what each function does. You would have to go in and read the code and try to make sense of it. Thats wasted time whereas if the function name was more descriptive you could clearly see exactly what the function is meant to do. Do you still not understand the difference between downloadOptimizedSvgs and downloadNewSvgFiles?

Reply:
okay, I get your point, I have renamed it into `downloadOptimizedSvgs` .